packages/eslint-plugin/src/rules/no-useless-template-expression.ts

```diff
@@ -0,0 +1,194 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import * as ts from 'typescript';
+
+import {
+  createRule,
+  getConstrainedTypeAtLocation,
+  getParserServices,
+  getStaticStringValue,
+  isTypeFlagSet,
+  isUndefinedIdentifier,
+} from '../util';
+
+export type NoUselessTemplateExpressionMessageId =
+  'noUselessTemplateExpression';
+
+export type NoUselessTemplateExpressionRuleObject = Parameters<
+  typeof createRule<[], NoUselessTemplateExpressionMessageId>
+>[0];
+
+export function makeNoUselessTemplateExpressionRuleObject(): NoUselessTemplateExpressionRuleObject {
+  // For some reason, TS isn't totally happy if you return the object directly,
+  // but it doesn't mind if you assign it to a variable first.
```

Comment:
[Refactor] This isn't the case on my end 🤔. I just get a lint complaint about `create` not having a return type. What errors were you seeing?

As a general trend, comments like _"For some reason, ..."_ are things we should try to hammer out before merge IMO.

Reply:
oh, yeah, I guess that lint warning is what i was referring to. Which is maybe a bug tbh? I don't get why I should have to put the return type there if I assign the object, but not if I return it.

Reply:
filed https://github.com/typescript-eslint/typescript-eslint/issues/8918